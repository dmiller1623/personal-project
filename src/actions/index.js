export const searchRelated = (searchedRelated) => ({
  type: 'SEARCH_RELATED',
  searchedRelated
}) 