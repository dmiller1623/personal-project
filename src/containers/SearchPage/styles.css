.search-page {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%; }

.searched-cards {
  overflow: scroll;
  flex-wrap: wrap;
  display: flex;
  width: 80%;
  height: 100vh; }
