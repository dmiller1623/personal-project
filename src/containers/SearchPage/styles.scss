.search-page {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  min-height: 100vh;
}

.searched-cards {
  flex-wrap: wrap;
  display: flex;
  width: 80%;
  // min-height: 100vh;
  background-color: blue;
}