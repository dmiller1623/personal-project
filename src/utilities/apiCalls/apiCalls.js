// import apiKey from '../../apiKey.js';
import { cleanData } from '../helper/helper.js';

export const getRelatedData = async (search) => {
  // const key = apiKey;
  const key = process.env.REACT_APP_API_KEY
  console.log(key)
  const url = `https://tastedive.com/api/similar?q=${search}&k=${key}&info=1`;
  console.log(url)
  try {
    const response = await fetch(url);
    const data = await response.json();
    const unresolvedData = cleanData(data);
    return Promise.all(unresolvedData);
  } catch (error) {
    throw new Error(error.message);
  }
};
