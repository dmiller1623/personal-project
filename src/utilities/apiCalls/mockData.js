export const mockSearchData = {
  Similar: {
    Info: [
      {
        Name: "The Doors",
        Type: "music",
        wTeaser: "\nThe Doors were an American rock band formed in 1965 in Los Angeles, with vocalist Jim Morrison, keyboardist Ray Manzarek, guitarist Robby Krieger, and John Densmore on drums. The band got its name, at Morrison's suggestion from the title of Aldous Huxley's book The Doors of Perception, which itself was a reference to a quote made by William Blake, \"If the doors of perception were cleansed, everything would appear to man as it is, infinite.\" They were unique and among the most controversial and influential rock acts of the 1960s, mostly because of Morrison's lyrics and charismatic but unpredictable stage persona. After Morrison's death in 1971, the remaining members continued as a trio until disbanding in 1973.",
        wUrl: "https://en.wikipedia.org/wiki/The_Doors",
        yUrl: "https://www.youtube-nocookie.com/embed/deB_u-to-IE",
        yID: "deB_u-to-IE"
      }
    ],
    Results: [
      {
        Name: "Jim Morrison",
        Type: "music",
        wTeaser: "\nJames Douglas Morrison (December 8, 1943 – July 3, 1971) was an American singer-songwriter and poet, best remembered as the lead vocalist of the Doors. Due to his poetic lyrics, distinctive voice, wild personality, performances, and the dramatic circumstances surrounding his life and early death, Morrison is regarded by music critics and fans as one of the most iconic and influential frontmen in rock music history.\nMorrison co-founded the Doors during the summer of 1965 in Venice, California. The band spent two years in obscurity until shooting to prominence with their #1 single in the United States, \"Light My Fire\", taken from their self-titled debut album. Morrison recorded a total of six studio albums with the Doors, all of which sold well and received critical acclaim. Though the Doors recorded two more albums after his death, the loss of Morrison was crippling to the band and they disbanded in 1973. In 1993, Morrison, as a member of the Doors, was inducted into the Rock and Roll Hall of Fame.\n",
        wUrl: "http://en.wikipedia.org/wiki/Jim_Morrisson",
        yUrl: "https://www.youtube-nocookie.com/embed/Yp5fe0Bzo1g",
        yID: "Yp5fe0Bzo1g"
      },
      {
        Name: "Janis Joplin",
        Type: "music",
        wTeaser: "\nJanis Lyn Joplin (January 19, 1943 – October 4, 1970), nicknamed  \"Pearl\",  was an American rock, soul and blues singer and songwriter, and one of the most successful and widely-known female rock stars of her era. After releasing three albums, she died of a heroin overdose at the age of 27. A fourth album, Pearl, was released in January 1971, just over three months after her death. It reached number one on the Billboard charts.\nIn 1967, Joplin rose to fame during an appearance at Monterey Pop Festival, where she was the lead singer of the then little-known San Francisco psychedelic rock band Big Brother and the Holding Company. After releasing two albums with the band, she left Big Brother to continue as a solo artist with her own backing groups, first the Kozmic Blues Band and then the Full Tilt Boogie Band. She appeared at the Woodstock festival and the Festival Express train tour. Five singles by Joplin went into the Billboard Hot 100, including a cover of the Kris Kristofferson song \"Me and Bobby McGee\", which reached number 1 in March 1971. Her most popular songs include her cover versions of \"Piece of My Heart\", \"Cry Baby\", \"Down on Me\", \"Ball and Chain\",  and \"Summertime\"; and her original song \"Mercedes Benz\", her final recording.",
        wUrl: "http://en.wikipedia.org/wiki/Janis_Joplin",
        yUrl: "https://www.youtube-nocookie.com/embed/7uG2gYE5KOs",
        yID: "7uG2gYE5KOs"
      }
    
    ]
  }
};