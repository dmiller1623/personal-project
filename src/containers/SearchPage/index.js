import React, { Component } from 'react'
import { connect } from 'react-redux';
import { getRelatedData } from '../../utilities/apiCalls/apiCalls';
import { searchRelated, addToResources } from '../../actions'
import RelatedItemsContainer from '../../components/RelatedItemsContainer'

export class SearchPage extends Component {
  constructor() {
    super();
    this.state = {
      subject: '',
      search: ''
    }
  }

  handleSubmit = async () => {
    const results = await getRelatedData(this.state.search)
    this.props.searchRelated(results)
  }

  handleChange = (event) => {
    const { name, value } = event.target
    this.setState({ [name]: value })
  }

  addResources = (resource) => {
    let additionalResourcesNames = this.props.additionalResources.map(resource => resource.Name)
    if(additionalResourcesNames.includes(resource.Name)) {
      return 
    } else {
      this.props.addToResources(resource)
    }
  } 

  render() {
    return(
      <div className="dropdown">
        <input className='subject-search'
          type='text'
          name='subject'
          onChange={this.handleChange}
        />
        <input className='search'
          type='text'
          name='search'
          onChange={this.handleChange}
        />
        <button onClick={this.handleSubmit}>Search</button>
        <RelatedItemsContainer relatedSearches={this.props.relatedSearches} addResources={this.addResources}/>
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  relatedSearches: state.relatedSearches,
  additionalResources: state.additionalResources
})

export const mapDispatchToProps = (dispatch) => ({
  searchRelated: (search) => dispatch(searchRelated(search)),
  addToResources: (resourceId) => dispatch(addToResources(resourceId))
})

export default connect(mapStateToProps, mapDispatchToProps)(SearchPage)