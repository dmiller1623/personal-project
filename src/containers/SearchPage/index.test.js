import React from 'react'
import { shallow } from 'enzyme'
import { mapStateToProps, mapDispatchToProps, SearchPage } from '../SearchPage'
import { searchRelated, addToResources } from '../../actions';
import { mockSearchData } from '../../utilities/apiCalls/mockData'


describe('SearchPage', () => {
  let wrapper
  it('should match the snapshot', () => {
    wrapper = shallow(<SearchPage store={{}}/>)
    expect(wrapper).toMatchSnapshot()
  })

  it('should call handle submit when a letter is pressed', () => {
    let eventObject = { target: { name: 'search', value: 'a' } }
    wrapper = shallow(<SearchPage store={{}}/>)

    wrapper.find('input').simulate('change', eventObject)
    expect(wrapper.state('search')).toEqual('a')
  })

  it('should call search related when clicking on submit', async () => {
    window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
      status: 200,
      json: () => Promise.resolve(mockSearchData)
    }));
    let searchRelatedMock = jest.fn()
    wrapper = shallow(<SearchPage searchRelated={searchRelatedMock}/>)
    
    await (wrapper).instance().handleSubmit()
    expect(searchRelatedMock).toHaveBeenCalled()

  })

  it('should call addToResources when add resources is called', () => {
    let mockResource = { name: 'jim morrison' }
    let mockAddtionalResources = []
    let mockAddToResources = jest.fn()
    wrapper = shallow(<SearchPage additionalResources={mockAddtionalResources} addToResources={mockAddToResources}/>)
    wrapper.instance().addResources(mockResource)

    expect(mockAddToResources).toHaveBeenCalledWith(mockResource)

  })

  it('should not call addToResources if the item already exsits', () => {
    let mockResource = { name: 'jim morrison' }
    let mockAddtionalResources = [{ name: 'jim morrison'}]
    let mockAddToResources = jest.fn()
    wrapper = shallow(<SearchPage additionalResources={mockAddtionalResources} addToResources={mockAddToResources}/>)
    wrapper.instance().addResources(mockResource)

    expect(mockAddToResources).not.toHaveBeenCalledWith(mockResource)

  })

  describe('mapStateToProps', () => {
    it('should return an array of related items', () => {
      const mockState = {
        relatedSearches: [],
        selectedItems: []
      }

      const expected = {
        relatedSearches: []
      }

      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    })
  })

  describe('mapDispatchToProps', () => {
    it('should call dispatch when using a function MDTP', () => {

      const mockDispatch = jest.fn();
      const actionToDispatch = searchRelated();
      const mappedProps = mapDispatchToProps(mockDispatch);
  
      mappedProps.searchRelated();
  
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    })

    it('should call dispatch when using a function MDTP', () => {

      const mockDispatch = jest.fn();
      const actionToDispatch = addToResources();
      const mappedProps = mapDispatchToProps(mockDispatch);
  
      mappedProps.addToResources();
  
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    })
  })
})