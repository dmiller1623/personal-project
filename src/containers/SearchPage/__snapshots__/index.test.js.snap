// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchPage
    store={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="header"
/>,
        <div>
          <input
                    className="search"
                    name="search"
                    onChange={[Function]}
                    placeholder="search for resources"
                    type="text"
          />
          <button
                    onClick={[Function]}
          >
                    Search
          </button>
          <NavLink
                    activeClassName="active"
                    aria-current="page"
                    to="/selectedResources"
          >
                    View Selected Resources
          </NavLink>
</div>,
        <div
          className="searched-cards"
>
          <RelatedItemsContainer
                    addResources={[Function]}
                    relatedSearches={undefined}
          />
</div>,
      ],
      "className": "search-page",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "header",
        },
        "ref": null,
        "rendered": null,
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="search"
              name="search"
              onChange={[Function]}
              placeholder="search for resources"
              type="text"
/>,
            <button
              onClick={[Function]}
>
              Search
</button>,
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/selectedResources"
>
              View Selected Resources
</NavLink>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "search",
              "name": "search",
              "onChange": [Function],
              "placeholder": "search for resources",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Search",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Search",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "View Selected Resources",
              "to": "/selectedResources",
            },
            "ref": null,
            "rendered": "View Selected Resources",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <RelatedItemsContainer
            addResources={[Function]}
            relatedSearches={undefined}
/>,
          "className": "searched-cards",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "addResources": [Function],
            "relatedSearches": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="header"
/>,
          <div>
            <input
                        className="search"
                        name="search"
                        onChange={[Function]}
                        placeholder="search for resources"
                        type="text"
            />
            <button
                        onClick={[Function]}
            >
                        Search
            </button>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        to="/selectedResources"
            >
                        View Selected Resources
            </NavLink>
</div>,
          <div
            className="searched-cards"
>
            <RelatedItemsContainer
                        addResources={[Function]}
                        relatedSearches={undefined}
            />
</div>,
        ],
        "className": "search-page",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "header",
          },
          "ref": null,
          "rendered": null,
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="search"
                name="search"
                onChange={[Function]}
                placeholder="search for resources"
                type="text"
/>,
              <button
                onClick={[Function]}
>
                Search
</button>,
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/selectedResources"
>
                View Selected Resources
</NavLink>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "search",
                "name": "search",
                "onChange": [Function],
                "placeholder": "search for resources",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Search",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Search",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "View Selected Resources",
                "to": "/selectedResources",
              },
              "ref": null,
              "rendered": "View Selected Resources",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <RelatedItemsContainer
              addResources={[Function]}
              relatedSearches={undefined}
/>,
            "className": "searched-cards",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "addResources": [Function],
              "relatedSearches": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
